docs/changelog.md

```diff
@@ -2,6 +2,11 @@
 
 This changelog documents user-relevant changes to the GitHub runner charm.
 
+## 2025-08-25
+
+- Refactor cleanup logic to one-shot delete from two-step approach.
```

Comment:
```suggestion
- Refactor cleanup logic to one-shot delete from two-step approach.
```

I think this is not really user-relevant, so shouldn't be part of the changelog.